File: project/compiler/jimmer-compiler-transactional/src/test/resources/transactional/ServiceATx.java
package org.babyfish.jimmer.sql.transaction;

import java.lang.Override;
import org.babyfish.jimmer.sql.JSqlClient;

@Component
public class ServiceATx extends ServiceA {
    ServiceATx(JSqlClient sqlClient) {
        super(sqlClient);
    }

    @Override
    public int a() {
        return sqlClient.transaction(Propagation.MANDATORY, () ->  {
            return super.a();
        });
    }

    @Override
    void b() {
        sqlClient.transaction(Propagation.REQUIRES_NEW, () ->  {
            super.b();
            return null;
        });
    }
}

File: project/compiler/jimmer-compiler/src/test/resources/transactional/ServiceATx.kt
@file:Suppress("warnings")

package org.babyfish.jimmer.sql.kt.transaction

import kotlin.Suppress
import kotlin.Unit
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.transaction.Propagation

@Component
public class ServiceATx(
    sqlClient: KSqlClient,
) : ServiceA(sqlClient) {
    override fun a(): Unit = this.sqlClient.transaction(Propagation.MANDATORY) {
        super.a()
    }

    internal override fun b(): Unit = this.sqlClient.transaction(Propagation.REQUIRES_NEW) {
        super.b()
    }
}
